prereise/gather/demanddata/transportation_electrification/mileage.py

```diff
@@ -0,0 +1,140 @@
+    raw_data = nhts_census[f"census_{census_region}_sorted"]
+    # Convert to data frame, adding column names
+    df = pd.DataFrame(raw_data, columns=const.nhts_census_column_names)
+
+    return df
+
+
+def remove_ldt(data: pd.DataFrame) -> pd.DataFrame:
+    """Remove light duty trucks from data loaded from nths_census.mat.
+
+    :param pd.DataFrame data: the data returned from load_data.
+    :return: (*pd.DataFrame*) -- the data loaded from load_data with all rows
+        involving LDT removed.
+    """
+
+    return data.loc[data["Vehicle type"] < 5].copy()
+
+
+# ldt = light duty truck
+# ldv = light duty vehicle
+def total_daily_vmt(
+    census_region: int,
+    comm_type: int,
+    locationstrategy: list[int],
+    input_day: list[int],
+) -> np.array[float]:
+    """load data and use the parameters to calculate total_daily_vmt.
+
+    :param int census_region: the type of census
+    :param int comm_type: the type of Commute
+    :param list locationstrategy: strategy for each location
+    :param list input_day: day of the week for each day in the year derived from
+        first_func
+    :return: (*list*) -- daily_vmt_total each row is a year of entries for each vehicle
+        type
+    """
+    # get the data
+    data = remove_ldt(load_data(census_region))
+    n = len(data)
+
+    if comm_type == 1:
+        # removes VMT for trips from work->home and home-> work
+        # (not exactly correct due to chained trips involving work and home but no
+        # way to remove the data)
+        if all([i != 2 for i in locationstrategy]):
+            locationstrategy = 2
+            # putting this part in function means rest of code won't be able to see
+            # change in locationstrategy value.
+            # will have to use global or return locationstrategy and reassign
+            print('"locationstrategy" changed to "Home and Work" for comm_type == 1')
+        # isolates home->work trips
+        trip_home_work = data.loc[data["why from"] == 1 and data["why to"] == 11]
+        # isolates work->home trips
+        trip_work_home = data.loc[data["why from"] == 11 and data["why to"] == 1]
+
+        # set trips from work to home to 0 distance
+        data.loc[trip_work_home.index, "Miles traveled"] = 0
+        data.loc[trip_home_work.index, "Miles traveled"] = 0
+
+    # pre-processing of vehicle trip data to determine weekday or weekend
+    # weekend in the case of 1 or 7
+    data.loc[data["Day of Week"].isin({1, 7}), "If Weekend"] = 1
+    # weekday in the case of 2-6 (inclusive)
+    data.loc[data["Day of Week"].isin(list(range(2, 7))), "If Weekend"] = 2
+
+    data_day = get_data_day(data)
+    daily_vmt_total = np.zeros(len(input_day))
+
+    for day_iter in range(len(input_day)):
+        for i in range(n):
+            if data_day[i] == input_day[day_iter]:
+                daily_vmt_total[day_iter][0] += data[i][12]
+                daily_vmt_total[day_iter][1] += 1

```

Comment:
It seems that these loops are still performing weekday/weekend grouping, without any month grouping.

Either way, this should be refactored to a two-step process, rather than nested loops:
1. Group trips (by either weekday/weekend alone, or by weekday/weekend and month), then count trips and sum mileage.
2. Map the collected information from (1) to the `daily_vmt_total` output based on the attributes of the days we want to return.

With nested loops, it seems we're performing the summing across all `data_day` entries _for each day we want to return data for_, so we're performing two orders of magnitude more computation than we need to.